app/gui/src/project-view/components/GraphEditor/widgets/WidgetTableEditor/TableHeader.vue

```diff
@@ -1,95 +1,96 @@
-  | { type: 'rowIndexColumn' }
+export interface ColumnSpecificParams {
+  columnParams:
+    | {
+        type: 'astColumn'
+        /** Setter called when column name is changed by the user. */
+        nameSetter: (newName: string) => void
+      }
+    | { type: 'newColumn'; enabled?: boolean; newColumnRequested: () => void }
+    | { type: 'rowIndexColumn' }
+}
 
 /**
- * Parameters recognized by this header component.
- *
- * They are set through `headerComponentParams` option in AGGrid column definition.
+ * General header parameters, set in default column configuration.
  */
-export type HeaderParams = ColumnSpecificHeaderParams & {
-  /**
-   * AgGrid mounts header components as separate "App", so we don't have access to any context.
-   * Threfore the tooltip registry must be provided by props.
-   */
-  tooltipRegistry: TooltipRegistry
+export interface HeaderParams {
+  /** The id of column whose header is currently edited. */
+  editedColId?: string | undefined
+  /** Callback called when editing this column is requested. */
+  onHeaderEditingStarted: (colId: string, revertChanges: () => void) => void
+  /** Callback called when editing of this column should be finished. */
+  onHeaderEditingStopped: (colId: string) => void
 }
 </script>
 
 <script setup lang="ts">
-const props = defineProps<{
-  params: IHeaderParams & HeaderParams
-}>()
-
-/** Re-provide tooltipRegistry. See `tooltipRegistry` docs in {@link HeaderParams} */
-provideTooltipRegistry.provideConstructed(props.params.tooltipRegistry)
-
-const editing = ref(false)
-const inputElement = ref<HTMLInputElement>()
-const editHandlers = computed(() =>
-  props.params.type === 'astColumn' ? props.params.editHandlers : undefined,
-)
+const props = defineProps<IHeaderParams & HeaderParams & ColumnSpecificParams>()
 
+const editing = computed(() => props.editedColId === props.column.getColId())
 watch(editing, (newVal) => {
-  if (newVal) {
-    editHandlers.value?.onHeaderEditingStarted?.((cancel: boolean) => {
-      if (cancel) editing.value = false
-      else acceptNewName()
-    })
-  } else {
-    editHandlers.value?.onHeaderEditingStopped?.()
+  if (!newVal) {
+    acceptNewName()
   }
 })
 
-watch(inputElement, (newVal, oldVal) => {
-  if (newVal != null && oldVal == null) {
-    // Whenever input field appears, focus and select text
-    newVal.focus()
-    newVal.select()
-  }
-})
+const inputElement = ref<HTMLInputElement>()
+
+function emitEditStart() {
+  props.onHeaderEditingStarted?.(props.column.getColId(), () => {
+    if (inputElement.value) {
+      inputElement.value.value = props.displayName
+    }
+  })
+}
+
+function emitEditEnd() {
+  props.onHeaderEditingStopped?.(props.column.getColId())
+}
+
+watch(
+  inputElement,
+  (newVal, oldVal) => {
+    if (newVal != null && oldVal == null) {
+      // Whenever input field appears, focus and select text
+      newVal.focus()
+      newVal.select()
+    }
+  },
+  { immediate: true },
+)
```

Comment:
I think this could be simpler:
- Why `immediate`? The `ref` will never have a value yet when this runs, right?
- Why check if the old value is nullish?